### ISSUE TITLE
rollup-plugin-ts Deprecation Error

### ISSUE THREAD

**Issue by Nimishxyz** [neutralinojs/neutralino.js#140]
## Problem
The `rollup-plugin-ts` package is deprecated and causes a build error related to JSON module assertions on newer node versions.

## Error Message
```
import * as compatData from '@mdn/browser-compat-data' assert { type: 'json' };
                                                       ^^^^^^

SyntaxError: Unexpected identifier 'assert'
```
## Steps to Reproduce
1. Run `npm install` with node >= 22
2. Execute `npm run build -- --dev`

## Recommended Solution from the devs
Replace `rollup-plugin-ts` with the following packages:
- `@rollup/plugin-typescript`
- `rollup-plugin-dts`

**Comment by ThisIsAdil** [comment]
after replacing and doing the respective changes it's showing this error
<img width="883" alt="Screenshot 2025-01-02 at 4 22 49 PM" src="https://github.com/user-attachments/assets/4ac2536f-5af0-41b0-ba30-c957d0d97b25" />

**Comment by Nimishxyz** [comment]
Yeah, I was getting the same error too and figured out that just changing the dependency isn't enough; it seems to have something to do with the config, I guess.

**Comment by ThisIsAdil** [comment]
Got it. Can you share some further insights? I'll try to debug this.

**Comment by Nimishxyz** [comment]
> ## Problem
> The `rollup-plugin-ts` package is deprecated and causes a build error related to JSON module assertions on newer node versions.
> 
> ## Error Message
> ```
> import * as compatData from '@mdn/browser-compat-data' assert { type: 'json' };
>                                                        ^^^^^^
> 
> SyntaxError: Unexpected identifier 'assert'
> ```
> 
> ## Steps to Reproduce
> 1. Run `npm install` with node >= 22
> 2. Execute `npm run build -- --dev`
> 
> ## Recommended Solution from the devs
> Replace `rollup-plugin-ts` with the following packages:
> 
> * `@rollup/plugin-typescript`
> * `rollup-plugin-dts`

for now replacing assert with `with` solves the issue

**Comment by ThisIsAdil** [comment]
Thank you for your response and for suggesting the solution to replace `assert` with `with`! I’m glad it resolves the issue. Should we consider this issue fully addressed and close it, or is there still something else I can assist with?v

**Comment by Julian-Sz** [comment]
@shalithasuranga this can be closed now
